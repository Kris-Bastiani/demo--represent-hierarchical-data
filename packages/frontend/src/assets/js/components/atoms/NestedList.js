import { h } from 'preact';

export default function NestedList({ items }) {
	return (<ul>
		{ items.map(item => {
			const hasChildren = item.children && item.children.length;

			return (<li key={item.name}>
				{ item.name }
				{ hasChildren ? <NestedList items={item.children} /> : null }
			</li>);
		}) }
	</ul>);
}
