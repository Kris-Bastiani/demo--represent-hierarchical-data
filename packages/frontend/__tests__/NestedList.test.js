import { h } from 'preact';
import { render } from '@testing-library/preact';

import mockEmployeesSortedByHierarchy from '../../../__mocks__/employeesSortedByHierarchy';

import NestedList from '../src/assets/js/components/atoms/NestedList';

describe('NestedList', () => {
	test('Matches snapshot', () => {
		const { container } = render(<NestedList items={mockEmployeesSortedByHierarchy} />);
		expect(container).toMatchSnapshot();
	});
});
