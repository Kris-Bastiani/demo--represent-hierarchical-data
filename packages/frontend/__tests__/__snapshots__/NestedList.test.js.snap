// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestedList Matches snapshot 1`] = `
<div>
  <ul>
    <li>
      Jamie
      <ul>
        <li>
          Alan
          <ul>
            <li>
              Martin
            </li>
            <li>
              Alex
            </li>
          </ul>
        </li>
        <li>
          Steve
          <ul>
            <li>
              David
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
